__tests__/useSelectedItems.test.tsx

```diff
@@ -0,0 +1,155 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+
+import useSelectedItems from "../src/index";
+
+const initialExampleItems = [
+  { id: 1, text: "What's up" },
+  { id: 2, text: "Hey there" },
+  { id: 3, text: "How you doin'" },
+];
+
+interface ExampleItem {
+  id: number;
+  text: string;
+}
+
+describe("useSelectedItems", () => {
+  it("should return all items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    const expectedItems = initialExampleItems.map((item) => ({
+      ...item,
+      isSelected: false,
+    }));
+
+    expect(result.current.items).toEqual(expectedItems);
+  });
+
+  it("should render with initialSelectedItems", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialSelectedItems: [initialExampleItems[0]],
+      initialItems: initialExampleItems,
+    }));
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.items).toContainEqual(
+      expect.objectContaining({ ...initialExampleItems[0], isSelected: true }),
+    );
```

Comment:
I would suggest to extract `initialExampleItems[0]` to a variable called `selectedItem`, since it can be reused for the assertions.

```suggestion
      initialSelectedItems: [selectedItem],
      initialItems: initialExampleItems,
    }));

    expect(result.current.selectedItems).toContainEqual(
      expect.objectContaining(selectedItem),
    );

    expect(result.current.items).toContainEqual(
      expect.objectContaining({ ...selectedItem, isSelected: true }),
    );
```